src/coreclr/src/pal/src/libunwind/CMakeLists.txt

```diff
@@ -1,21 +1,125 @@
-add_subdirectory(src)
-
 # define variables for the configure_file below
 
-if (CLR_CMAKE_HOST_ARCH_AMD64)
-    set(arch x86_64)
-elseif(CLR_CMAKE_HOST_ARCH_ARM64)
-    set(arch aarch64)
-elseif(CLR_CMAKE_PLATFROM_ARCH_ARM)
-    set(arch arm)
-elseif(CLR_CMAKE_HOST_ARCH_I386)
-    set(arch x86)
-endif()
-
 set(PKG_MAJOR "1")
 set(PKG_MINOR "5")
 set(PKG_EXTRA "-rc2")
+if(CLR_CMAKE_HOST_UNIX)
+    if (CLR_CMAKE_HOST_ARCH_AMD64)
+      set(arch x86_64)
+    elseif(CLR_CMAKE_HOST_ARCH_ARM64)
+      set(arch aarch64)
+    elseif(CLR_CMAKE_PLATFROM_ARCH_ARM)
+      set(arch arm)
+    elseif(CLR_CMAKE_HOST_ARCH_I386)
+        set(arch x86)
+    endif ()
+endif(CLR_CMAKE_HOST_UNIX)
+
+if(CLR_CMAKE_HOST_WIN32)
+    if (CLR_CMAKE_TARGET_ARCH_AMD64)
+      set(TARGET_AMD64 1)
+      set(arch x86_64)
+      add_definitions(-D__x86_64__)
+      add_definitions(-D__amd64__)
+    elseif(CLR_CMAKE_TARGET_ARCH_ARM64)
+      set(TARGET_AARCH64 1)
+      set(arch aarch64)
+      add_definitions(-D__aarch64__)
+    elseif(CLR_CMAKE_TARGET_ARCH_ARM)
+      set(TARGET_ARM 1)
+      set(arch arm)
+      add_definitions(-D__arm__)
+    else ()
+      message(FATAL_ERROR "Unrecognize value in environment variable TARGET")
+    endif ()
+
+    set(PACKAGE_STRING "libunwind")
+    set(PACKAGE_BUGREPORT "")
+
+    include(CheckCSourceCompiles)
+    include(CheckIncludeFiles)
+
+    # Our posix abstraction layer will provide these headers
+    set(HAVE_ELF_H 1)
+    set(HAVE_ENDIAN_H 1)
+
+    # MSVC compiler is currently missing C11 stdalign.h header
+    # Fake it until support is added
+    check_include_files(stdalign.h HAVE_STDALIGN_H)
+    if (NOT HAVE_STDALIGN_H)
+        configure_file(include/win/fakestdalign.h.in ${CMAKE_CURRENT_BINARY_DIR}/include/stdalign.h)
+    endif (NOT HAVE_STDALIGN_H)
+
+    # MSVC compiler is currently missing C11 stdatomic.h header
+    # Fake it until support is added
+    check_include_files(stdatomic.h HAVE_STDATOMIC_H)
+    if (NOT HAVE_STDATOMIC_H)
+        configure_file(include/win/fakestdatomic.h.in ${CMAKE_CURRENT_BINARY_DIR}/include/stdatomic.h)
+    endif (NOT HAVE_STDATOMIC_H)
+
+    # MSVC compiler is currently missing C11 _Thread_local
+    check_c_source_compiles("void main() { _Thread_local int a; }"  HAVE_THREAD_LOCAL)
+    if (NOT HAVE_THREAD_LOCAL)
+        add_definitions(-D_Thread_local=)
```

Comment:
In `C`, `thread_local` is defined as `_Thread_local` by thread.h.  Support for both seem to be missing from MSVC when compiled as `C`.

Since we are only using this for DAC unwind, I assumed thread_local didn't matter, but I might be proven wrong.